agera/src/main/java/com/google/android/agera/BaseObservable.java

```diff
@@ -38,149 +39,131 @@
  */
 public abstract class BaseObservable implements Observable {
   @NonNull
-  private final Worker worker;
+  private static final Object[] NO_UPDATABLES_OR_HANDLERS = new Object[0];
+  @NonNull
+  private final WorkerHandler handler;
+  private final int shortestUpdateWindowMillis;
+
+  @NonNull
+  private Object[] updatablesAndHandlers;
+  private int size;
+  private long lastUpdateTimestamp;
 
   protected BaseObservable() {
+    this(0);
+  }
+
+  BaseObservable(final int shortestUpdateWindowMillis) {
     checkState(Looper.myLooper() != null, "Can only be created on a Looper thread");
-    worker = new Worker(this);
+    this.shortestUpdateWindowMillis = shortestUpdateWindowMillis;
+    this.handler = workerHandler();
+    this.updatablesAndHandlers = NO_UPDATABLES_OR_HANDLERS;
+    this.size = 0;
   }
 
   @Override
-  public final void addUpdatable(@NonNull final Updatable updatable) {
+  public synchronized final void addUpdatable(@NonNull final Updatable updatable) {
```

Comment:
I would prefer a private object for locking (`synchronized` on). That was one (implicit) responsibility of the Worker instance before this commit.